File: app/src/main/java/com/example/pcmall/service/OrderService.java
package com.example.pcmall.service;

import com.example.pcmall.model.Order;
import com.example.pcmall.model.response.ResponseResult;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface OrderService {
    @FormUrlEncoded
    @POST("searchOrderList")
    Observable<ResponseResult<List<Order>>> searchOrderList(
            @Field("searchValue") String searchValue,
            @Field("uid") String uid,
            @Field("type") Integer type,
            @Field("currentPage") Integer currentPage,
            @Field("pageSize") Integer pageSize);

    @FormUrlEncoded
    @POST("getRecordsFiltered")
    Observable<ResponseResult<Long>> getRecordsFiltered(
            @Field("searchValue") String searchValue,
            @Field("uid") String uid,
            @Field("type") Integer type);

    @FormUrlEncoded
    @POST("createOrder")
    Observable<ResponseResult<String>> createOrder(@Field("uid") String uid, @Field("aid") Integer aid);

    @FormUrlEncoded
    @POST("payOrder")
    Observable<ResponseResult<String>> payOrder(@Field("oid") String oid);

    @FormUrlEncoded
    @POST("finishOrder")
    Observable<ResponseResult<String>> finishOrder(@Field("oid") String oid);

    @FormUrlEncoded
    @POST("cancelOrder")
    Observable<ResponseResult<String>> cancelOrder(@Field("oid") String oid);

    @FormUrlEncoded
    @POST("refundOrder")
    Observable<ResponseResult<String>> refundOrder(@Field("oid") String oid);
}

File: core/network/src/main/java/com/example/pcmallcompose/core/network/service/OrderService.kt
package com.example.pcmallcompose.core.network.service

import com.example.pcmallcompose.core.model.Order
import com.example.pcmallcompose.core.model.response.ResponseResult
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

interface OrderService {
    @FormUrlEncoded
    @POST("searchOrderList")
    suspend fun searchOrderList(
        @Field("searchValue") searchValue: String,
        @Field("uid") uid: String,
        @Field("type") type: Int,
        @Field("currentPage") currentPage: Int,
        @Field("pageSize") pageSize: Int
    ): ResponseResult<List<Order>>

    @FormUrlEncoded
    @POST("getRecordsFiltered")
    suspend fun getRecordsFiltered(
        @Field("searchValue") searchValue: String,
        @Field("uid") uid: String,
        @Field("type") type: Int
    ): ResponseResult<Long>

    @FormUrlEncoded
    @POST("createOrder")
    suspend fun createOrder(
        @Field("uid") uid: String,
        @Field("aid") aid: Int
    ): ResponseResult<String>

    @FormUrlEncoded
    @POST("payOrder")
    suspend fun payOrder(@Field("oid") oid: String): ResponseResult<String>

    @FormUrlEncoded
    @POST("finishOrder")
    suspend fun finishOrder(@Field("oid") oid: String): ResponseResult<String>

    @FormUrlEncoded
    @POST("cancelOrder")
    suspend fun cancelOrder(@Field("oid") oid: String): ResponseResult<String>

    @FormUrlEncoded
    @POST("refundOrder")
    suspend fun refundOrder(@Field("oid") oid: String): ResponseResult<String>
}